Segfault after ncurses update to 6.0

The distro is gentoo. I had quite a huge update and now tig crashes.

I downloaded from the repo, my commit SHA is cef17fcab8f8a95a257d9ee15bae271912c81906.
I can rebuild probably any other dependency with more debug symbols if required.

    #0  0x00007fd36c8e9896 in wchgat () from /lib64/libncurses.so.6
    #1  0x000000000040fee4 in set_view_attr (type=LINE_PP_REFS, view=0x642940 <diff_view>) at src/draw.c:46
    #2  draw_chars (view=view@entry=0x642940 <diff_view>, type=type@entry=LINE_PP_REFS, 
        string=string@entry=0x650080 <text> "Refs: [master], {origin/master}, {origin/HEAD}", 
        max_width=max_width@entry=118, use_tilde=use_tilde@entry=false, length=-1) at src/draw.c:73
    #3  0x000000000041019c in draw_text_expanded (view=view@entry=0x642940 <diff_view>, type=LINE_PP_REFS, 
        string=string@entry=0x1b40500 "Refs: [master], {origin/master}, {origin/HEAD}", length=46, max_width=118, 
        use_tilde=use_tilde@entry=false) at src/draw.c:118
    #4  0x0000000000410b77 in draw_textn (length=<optimized out>, 
        string=0x1b40500 "Refs: [master], {origin/master}, {origin/HEAD}", type=<optimized out>, 
        view=0x642940 <diff_view>) at src/draw.c:131
    #5  view_column_draw (view=0x642940 <diff_view>, line=<optimized out>, lineno=1) at src/draw.c:552
    #6  0x00000000004112bd in draw_view_line (view=view@entry=0x642940 <diff_view>, lineno=lineno@entry=1)
        at src/draw.c:633
    #7  0x0000000000411645 in redraw_view_from (view=0x642940 <diff_view>, lineno=1) at src/draw.c:677
    #8  0x000000000041798e in update_view (view=view@entry=0x642940 <diff_view>) at src/view.c:623
    #9  0x0000000000414b1f in get_input (prompt_position=prompt_position@entry=-1, key=key@entry=0x7fffd7552310)
        at src/display.c:583
    #10 0x000000000041176e in prompt_input (prompt=prompt@entry=0x42e0b2 "", input=input@entry=0x640c40 <incremental>)
        at src/prompt.c:49
    #11 0x0000000000412c55 in read_prompt_incremental (prompt=prompt@entry=0x42e0b2 "", edit_mode=edit_mode@entry=false, 
        allow_empty=allow_empty@entry=false, handler=handler@entry=0x404250 <key_combo_handler>, 
        data=data@entry=0x7fffd7552770) at src/prompt.c:178
    #12 0x000000000040464a in read_key_combo (keymap=<optimized out>) at src/tig.c:649
    #13 0x000000000040381a in main (argc=<optimized out>, argv=<optimized out>) at src/tig.c:712

I do a fresh clone and configure. When I try to download & make, I get an error:

```
pzie:~/git/$ git clone https://github.com/jonas/tig.git tig-clean
Cloning into 'tig-clean'...
remote: Counting objects: 11725, done.
remote: Total 11725 (delta 0), reused 0 (delta 0), pack-reused 11724
Receiving objects: 100% (11725/11725), 4.93 MiB | 1.60 MiB/s, done.
Resolving deltas: 100% (8075/8075), done.
pzie:~/git/$ cd tig-clean
[tig-clean] pzie:pzie/git/tig-clean/$ make
        CC  src/tig.o
In file included from src/tig.c:16:0:
include/tig/tig.h:77:4: warning: #warning SysV or X/Open-compatible Curses installation is required. [-Wcpp]
 #  warning SysV or X/Open-compatible Curses installation is required.
    ^
include/tig/tig.h:78:4: warning: #warning Will assume Curses is found in default include and library path. [-Wcpp]
 #  warning Will assume Curses is found in default include and library path.
    ^
include/tig/tig.h:79:4: warning: #warning To fix any build issues please use autotools to configure Curses. [-Wcpp]
 #  warning To fix any build issues please use autotools to configure Curses.
    ^
include/tig/tig.h:80:4: warning: #warning See INSTALL.adoc file for instructions. [-Wcpp]
 #  warning See INSTALL.adoc file for instructions.
    ^
        CC  src/types.o
        CC  src/string.o
        CC  src/util.o
        CC  src/map.o
        CC  src/argv.o
        CC  src/io.o
        CC  src/refdb.o
       GEN  src/builtin-config.c
        CC  src/builtin-config.o
        CC  src/request.o
        CC  src/line.o
        CC  src/keys.o
        CC  src/repo.o
        CC  src/options.o
        CC  src/draw.o
        CC  src/prompt.o
        CC  src/display.o
        CC  src/view.o
        CC  src/search.o
        CC  src/parse.o
        CC  src/watch.o
        CC  src/pager.o
        CC  src/log.o
        CC  src/diff.o
        CC  src/help.o
        CC  src/tree.o
        CC  src/blob.o
        CC  src/blame.o
        CC  src/refs.o
        CC  src/status.o
        CC  src/stage.o
        CC  src/main.o
        CC  src/stash.o
        CC  src/grep.o
        CC  src/ui.o
        CC  src/graph.o
        CC  src/graph-v1.o
        CC  src/graph-v2.o
        CC  compat/hashtab.o
      LINK  src/tig
/usr/lib/gcc/x86_64-pc-linux-gnu/4.9.4/../../../../x86_64-pc-linux-gnu/bin/ld: src/display.o: undefined reference to symbol 'erasechar'
/lib64/libtinfo.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```


After that I run `make configure` and `./configure --prefix=$mydir`, output attached.

`make; make install` goes normally. Running tig gives me a segfault.

--- a/tools/ax_lib_readline.m4
+++ b/tools/ax_lib_readline.m4
@@ -34,16 +34,19 @@ fi
 AC_CACHE_VAL(bash_cv_termcap_lib,
 [AC_CHECK_FUNC(tgetent, bash_cv_termcap_lib=libc,
 if test "$ax_cv_curses_which" = "ncursesw"; then
-	[AC_CHECK_LIB(ncursesw, tgetent, bash_cv_termcap_lib=libncursesw)]
+	[AC_CHECK_LIB(ncursesw, tgetent, bash_cv_termcap_lib=libncursesw,
+		[AC_CHECK_LIB(tinfow, tgetent, bash_cv_termcap_lib=libtinfow)]
+	)]
 elif test "$ax_cv_curses_which" = "ncurses"; then
-	[AC_CHECK_LIB(ncurses, tgetent, bash_cv_termcap_lib=libncurses)]
+	[AC_CHECK_LIB(ncurses, tgetent, bash_cv_termcap_lib=libncurses,
+		[AC_CHECK_LIB(tinfo, tgetent, bash_cv_termcap_lib=libtinfo)]
+	)]
 elif test "$ax_cv_curses_which" = "plaincurses"; then
 	[AC_CHECK_LIB(curses, tgetent, bash_cv_termcap_lib=libcurses)]
 else
 	[AC_CHECK_LIB(termcap, tgetent, bash_cv_termcap_lib=libtermcap,
-		[AC_CHECK_LIB(tinfo, tgetent, bash_cv_termcap_lib=libtinfo,
-			bash_cv_termcap_lib=gnutermcap
-	)])]
+		bash_cv_termcap_lib=gnutermcap
+	)]
 fi
 )])
 if test "X$_bash_needmsg" = "Xyes"; then
@@ -56,6 +59,9 @@ TERMCAP_LIB="./lib/termcap/libtermcap.a"
 TERMCAP_DEP="./lib/termcap/libtermcap.a"
 elif test $bash_cv_termcap_lib = libtermcap && test -z "$prefer_curses"; then
 TERMCAP_LIB=-ltermcap
+TERMCAP_DEP=
+elif test $bash_cv_termcap_lib = libtinfow; then
+TERMCAP_LIB=-ltinfow
 TERMCAP_DEP=
 elif test $bash_cv_termcap_lib = libtinfo; then
 TERMCAP_LIB=-ltinfo

